src/main/java/com/example/weatherwaveapi/config/ApplicationConfiguration.java

```diff
@@ -0,0 +1,24 @@
+package com.example.weatherwaveapi.config;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.boot.web.client.RestTemplateBuilder;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.web.client.RestTemplate;
+
+import java.time.Duration;
+
+
+@Configuration
+@RequiredArgsConstructor
+public class ApplicationConfiguration {
+    private final GeneralSettings generalSettings;
+
+    @Bean
+    public RestTemplate restTemplate(RestTemplateBuilder builder) {
+        return builder
+                .setConnectTimeout(Duration.ofMillis(generalSettings.getRestTemplateProperties().connectionTimeout()))
+                .setReadTimeout(Duration.ofMillis(generalSettings.getRestTemplateProperties().readTimeout()))
+                .build();
+    }
+}
```

Comment:
please pull last changes from the remote repo, this Configuration already implemented

Reply:
<img width="963" alt="image" src="https://github.com/ann-sargsyan/WeatherWaveAPI/assets/108473070/ef59fb0c-b197-4f31-a8a4-d4ee3dc95a83">